mono/metadata/cominterop.c

```diff
@@ -3664,3 +3664,54 @@ ves_icall_System_Runtime_InteropServices_Marshal_FreeBSTR (gpointer ptr)
 {
 	mono_free_bstr (ptr);
 }
+
+void*
+mono_cominterop_get_com_interface (MonoObject* object, MonoClass* ic)
+{
+#ifndef DISABLE_COM
+	MonoError error;
+
+	if (!object)
+		return NULL;
+
+	if (cominterop_object_is_rcw (object)) {
+		MonoClass *klass = NULL;
+		MonoRealProxy* real_proxy = NULL;
+		if (!object)
+			return NULL;
+		klass = mono_object_class (object);
+		if (!mono_class_is_transparent_proxy (klass)) {
+			g_assert_not_reached ();
+			return NULL;
+		}
+
+		real_proxy = ((MonoTransparentProxy*)object)->rp;
+		if (!real_proxy) {
+			g_assert_not_reached ();
+			return NULL;
+		}
+
+		klass = mono_object_class (real_proxy);
+		if (klass != mono_class_get_interop_proxy_class ()) {
+			g_assert_not_reached ();
+			return NULL;
+		}
+
+		if (!((MonoComInteropProxy*)real_proxy)->com_object) {
+			g_assert_not_reached ();
+			return NULL;
+		}
+
+		void* com_itf = cominterop_get_interface_checked (((MonoComInteropProxy*)real_proxy)->com_object, ic, &error);
+		mono_error_set_pending_exception (&error);
+		return com_itf;
+	}
+	else {
+		void* ccw_entry = cominterop_get_ccw_checked (object, ic, &error);
+		mono_error_set_pending_exception (&error);
```

Comment:
This function sets a global variable that will result in some other thread throwing this exception without warning. It's useful when in icalls and specific contexts, but it's general use is discouraged for new code. Could you consider how to handle this error in this code, or in the callers of this code?